ed_sensor_integration/tools/fitter_viz.cpp

```diff
@@ -0,0 +1,382 @@
+    {
+        for (unsigned int j=0; j < shape[i].size()-1; j++)
+        {
+            drawLine(canvas, shape[i][j], shape[i][j+1], pose, resolution, origin_x, origin_y, color);
+        }
+        drawLine(canvas, shape[i][0], shape[i][shape[0].size()-1], pose, resolution, origin_x, origin_y, color);
+    }
+
+    // paint entity center
+
+    int x_p_ent = origin_x + (int)(pose.t.x * resolution);
+    int y_p_ent = origin_y - (int)(pose.t.y * resolution);
+
+    cv::Point Entity_center(x_p_ent, y_p_ent);
+    cv::circle(canvas, Entity_center, 3, color, cv::FILLED);
+
+}
+// ----------------------------------------------------------------------------------------------------
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "ed_fitter");
+    ros::NodeHandle nh;
+
+    if (argc != 4)
+    {
+        usage();
+        return 1;
+    }
+
+    std::string model_name = argv[2];
+
+    ed::WorldModel world_model;
+    if (!loadWorldModel(model_name, world_model)){
+        std::cerr << "World model '" << model_name << "' could not be loaded." << std::endl;
+        return 1;
+    }
+
+    std::string entity_id = argv[3];
+    ed::EntityConstPtr e = world_model.getEntity(entity_id);
+    if (!e)
+    {
+        std::cerr << "Entity '" << entity_id << "' could not be found in world model '" << model_name << "'." << std::endl;
+        return 1;
+    }
+
+    tue::filesystem::Path path = argv[1];
+    if (!path.exists())
+    {
+        std::cerr << "Path '" << path << "' does not exist." << std::endl;
+        return 1;
+    }
+
+    tue::filesystem::Crawler crawler;
+
+    if (path.isDirectory())
+        crawler.setRootPath(path);
+    else
+        crawler.setRootPath(path.parentPath());
+
+    // visualization parameters
+    int canvas_width = 600; //pixels
+    int canvas_height = 600; // pixels
+    float canvas_resolution = 100; //pixels per meter
+
+    int sensor_x = canvas_width/2;
+    int sensor_y = canvas_height * 8/10;
+    cv::Point sensorlocation(sensor_x, sensor_y);
+
+    cv::Scalar sensor_colour(0, 255, 0); // green
+    cv::Scalar entity_colour(0, 255, 255); // yellow
+    cv::Scalar fitted_colour(243, 192, 15); // blue
+    cv::Scalar measurement_colour(161, 17, 187); // purple
+
+    Updater updater;
+    Fitter fitter;
+
+    std::vector<Snapshot> snapshots;
+    unsigned int i_snapshot = 0;
+
+    while(ros::ok())
+    {
+        if (i_snapshot >= snapshots.size())
+        {
+            bool file_found = true;
+            tue::filesystem::Path filename;
+
+            if (path.isRegularFile() && snapshots.empty())
+                filename = path;
+            else
+            {
+                file_found = false;
+                while (crawler.nextPath(filename))
+                {
+                    if (filename.extension() == ".json")
+                    {
+                        file_found = true;
+                        break;
+                    }
+                }
+            }
```

Comment:
Unclear what this logic is doing. I got the feeling, this can be much more simple. And comments would also be really helpful.

Reply:
It is legacy code that should be treated by #52